pkg/R/DataFrame.R

```diff
@@ -253,6 +253,96 @@ setMethod("unpersist",
             x
           })
 
+#' Repartition
+#'
+#' Return a new DataFrame that has exactly numPartitions partitions.
+#'
+#' @param x A SparkSQL DataFrame
+#' @param numPartitions The number of partitions to use.
+#' @rdname repartition
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' sqlCtx <- sparkRSQL.init(sc)
+#' path <- "path/to/file.json"
+#' df <- jsonFile(sqlCtx, path)
+#' newDF <- repartition(df, 2L)
+#'}
+
+setGeneric("repartition", function(x, numPartitions) { standardGeneric("repartition") })
+
+#' @rdname repartition
+#' @export
+setMethod("repartition",
+          signature(x = "DataFrame", numPartitions = "integer"),
```

Comment:
I think that's a great idea. Do you happen to know how to grab the name of an incoming argument? I think the utility function could look like this:
```R
numToInt <- function(num) {
  if (!is.integer(num)) {
    warning(paste("Coercing", num, "to integer."))
    as.integer(num)
  }
}
```
The only problem is that `num` in the `warning()` function converts to the value, not the name, so we get:
```R
> numPartitions <- 5
> numToInt(numPartitions)
[1] 5
Warning message:
In numToInt(numPartitions) : Coercing 5 to integer.
```
I think it's ideal to have the message print the name so that we could use it as a true utility function and have the warning message adapt accordingly. This way the warning message will make sense relative to the function it's used in.

Ideally we want `Coercing numPartitions to integer.` in this case.